addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -455,10 +429,10 @@ static func get_general_blocks() -> Array[Block]:
 	b.block_format = "Load file {file_path: STRING} as sound {name: STRING}"
 	b.statement = (
 		"""
-		VAR_DICT[{name}] = AudioStreamPlayer.new()
-		VAR_DICT[{name}].name = {name}
-		VAR_DICT[{name}].set_stream(load({file_path}))
-		add_child(VAR_DICT[{name}])
+		var sound__ = AudioStreamPlayer.new()
+		sound__.name = {name}
```

Comment:
I think that's a good idea even though with the suffixed counter it's unlikely to clash. Still, I think the common convention is that you _prefix_ internal variables with `_` or `__`. Furthermore, we could add in something semi-unique like `bcp` (for Block Code Plugin). So, `__bcp_<something>`.

In fact, if this variable isn't going to be exposed to the user, then there's no need to name it like the thing it's for at all. So, you just declare all the internal variables `__bcp`. The generator recognizes that (without tricky regex matching or trimming any parts) and suffixes the generated variable with a `<counter>` or `_<counter>`.

Reply:
I think the whole point of generating the script like this is to have it exposed to the user, otherwise we could just have a library function like `Lib.play_sound()` that would bypass the need for local variables anyways.

That's also why I implemented a counter, instead of having a unique ID appended to each variable, because it makes the code look a lot nicer. Anyways, a prefix will definitely help here since we can also prevent the user from creating variables with our prefix.

Reply:
That's fair. I think just `_` or `__` prefix is good enough to make conflict with a user supplied variable extremely unlikely.